examples/tablet.c

```diff
@@ -164,20 +162,33 @@ static void handle_pad_ring(struct tablet_pad_state *pstate,
 	}
 }
 
+static void handle_input_add(struct compositor_state *cstate,
+		struct wlr_input_device *inputdev) {
+	struct sample_state *sample = cstate->data;
+	if (inputdev->type == WLR_INPUT_DEVICE_TABLET_TOOL) {
+		sample->width_mm = inputdev->width_mm == 0 ?
+			20 : inputdev->width_mm;
+		sample->height_mm = inputdev->height_mm == 0 ?
+			10 : inputdev->height_mm;
+	}
+}
+
 int main(int argc, char *argv[]) {
 	wlr_log_init(L_DEBUG, NULL);
 	struct sample_state state = {
 		.tool_color = { 1, 1, 1, 1 },
 		.pad_color = { 0.5, 0.5, 0.5, 1.0 }
 	};
-	struct compositor_state compositor = { 0,
+	struct compositor_state compositor = {
 		.data = &state,
 		.output_frame_cb = handle_output_frame,
 		.tool_axis_cb = handle_tool_axis,
 		.tool_proximity_cb = handle_tool_proximity,
 		.tool_button_cb = handle_tool_button,
 		.pad_button_cb = handle_pad_button,
 		.pad_ring_cb = handle_pad_ring,
+		.input_add_cb = handle_input_add,
+		0
```

Comment:
(is that 0 needed at all? anything not specified should be zeroed out by construction)